HealthcareSystemFrontends/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -491,6 +511,43 @@ private void ListViewExtendMenu_SelectionChanged(object sender, SelectionChanged
             }
         }
 
+        private void UpdateMenuOptionSelectionHistory(int currentlySelectedMenuOptionIndex)
+        {
+            if (_selectedMenuOptionsHistory.Count >= 0 && _selectedMenuOptionsHistory.Count < 2)
+            {
+                _selectedMenuOptionsHistory.Add(currentlySelectedMenuOptionIndex);
+            }
+            else
+            {
+                _selectedMenuOptionsHistory.RemoveAt(0);
+                _selectedMenuOptionsHistory.Add(currentlySelectedMenuOptionIndex);
+
+                PreviousSelectedMenuOptionIndex = _selectedMenuOptionsHistory[0];
+                SelectedMenuOptionIndex = _selectedMenuOptionsHistory[1];
+            }
```

Comment:
Could you explain why this method looks like this? Is the goal to limit the number of remembered items, or? I feel like there must be a nicer way to do that.

Reply:
List _selectedMenuOptionsHistory always contains indexes of the last two selected side panels. Initially, that list is empty, so firstly I just add indexes to the list. Later, on each selection change of side panel index, I just remove the first item from the list because that is now the oldest index, therefor previously selected index will move to the first position in the list and the currently selected will be on the second position in the list. So list always contains just two items: Previously and Currently selected index.

Reply:
I found what the issue with just using the SelectedMenuItemIndex and adding a field for just the previous index was. I made two suggestions for alternate implementations with that issue fixed, you can check them out over on PR #124 and PR #125. Feel free to do merge or close those PRs as you see fit.

Reply:
Thank you very much for your time and advice! 😄